lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -1216,6 +1216,13 @@ def generate_module_table(type, search_term = nil) # :nodoc:
             )
           end
 
+          def generate_post_list(type,search_term=nil)
```

Comment:
This command_dispatcher is designed for use by the console UI, I would suggest looking more into what this dispatcher interacts with, and how to make that available to your server object instead of trying to extend here.  This data format is presentation layer vs data layer, you are going to need to format the objects into json that you feed the session-ui anyways so it may be more efficient to obtain it more directly.

Reply:
Thank you, sir, This helps a lot. I am using framework.post.keys to get a list of post mods.